test/functional/invalidblockrequest.py

```diff
@@ -11,68 +11,64 @@
 re-requested.
 """
 import copy
-import time
 
 from test_framework.blocktools import create_block, create_coinbase, create_transaction
-from test_framework.comptool import RejectResult, TestInstance, TestManager
 from test_framework.messages import COIN
-from test_framework.mininode import network_thread_start
-from test_framework.test_framework import ComparisonTestFramework
+from test_framework.mininode import network_thread_start, P2PStub
+from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import assert_equal
 
-class InvalidBlockRequestTest(ComparisonTestFramework):
+class InvalidBlockRequestTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
         self.setup_clean_chain = True
+        self.extra_args = [["-whitelist=127.0.0.1"]]
 
     def run_test(self):
-        test = TestManager(self, self.options.tmpdir)
-        test.add_all_connections(self.nodes)
-        self.tip = None
-        self.block_time = None
+        # Add p2p connection to node0
+        node = self.nodes[0]  # convenience reference to the node
+        node.add_p2p_connection(P2PStub())
+
         network_thread_start()
-        test.run()
+        node.p2p.wait_for_verack()
 
-    def get_tests(self):
-        if self.tip is None:
-            self.tip = int("0x" + self.nodes[0].getbestblockhash(), 0)
-        self.block_time = int(time.time()) + 1
+        best_block = node.getblock(node.getbestblockhash())
+        tip = int("0x" + node.getbestblockhash(), 0)
+        height = best_block["height"] + 1
+        block_time = best_block["time"] + 1
 
         self.log.info("Create a new block with an anyone-can-spend coinbase")
 
         height = 1
-        block = create_block(self.tip, create_coinbase(height), self.block_time)
-        self.block_time += 1
+        block = create_block(tip, create_coinbase(height), block_time)
+        block_time += 1
         block.solve()
         # Save the coinbase for later
-        self.block1 = block
-        self.tip = block.sha256
+        block1 = block
+        tip = block.sha256
         height += 1
-        yield TestInstance([[block, True]])
+        node.p2p.send_blocks_and_test([block1], node, True)
 
         self.log.info("Mature the block.")
+        node.generate(100)
 
-        test = TestInstance(sync_every_block=False)
-        for i in range(100):
-            block = create_block(self.tip, create_coinbase(height), self.block_time)
-            block.solve()
-            self.tip = block.sha256
-            self.block_time += 1
-            test.blocks_and_transactions.append([block, True])
-            height += 1
-        yield test
+        best_block = node.getblock(node.getbestblockhash())
+        tip = int("0x" + node.getbestblockhash(), 0)
```

Comment:
In commit "Change invalidblockrequest to use BitcoinTestFramework"

Maybe avoid 0x prefix here and above by calling int(hash, 16)